From 28b2d58efa42fea84c7b7eb6f2807cf33f689827 Mon Sep 17 00:00:00 2001
From: jjm2473 <jjm2473@gmail.com>
Date: Sat, 8 Feb 2025 16:30:52 +0800
Subject: [PATCH] Fix class_create api changed

---
 drivers/video/rockchip/mpp/mpp_service.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/rockchip/mpp/mpp_service.c b/drivers/video/rockchip/mpp/mpp_service.c
index e523fd1..642e159 100644
--- a/drivers/video/rockchip/mpp/mpp_service.c
+++ b/drivers/video/rockchip/mpp/mpp_service.c
@@ -19,6 +19,7 @@
 #include <linux/slab.h>
 #include <linux/nospec.h>
 #include <linux/mfd/syscon.h>
+#include <linux/version.h>
 
 #include "mpp_debug.h"
 #include "mpp_common.h"
@@ -361,7 +362,11 @@ static int mpp_service_probe(struct platform_device *pdev)
 	atomic_set(&srv->shutdown_request, 0);
 	platform_set_drvdata(pdev, srv);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
 	srv->cls = class_create(THIS_MODULE, MPP_CLASS_NAME);
+#else
+	srv->cls = class_create(MPP_CLASS_NAME);
+#endif
 	if (PTR_ERR_OR_ZERO(srv->cls))
 		return PTR_ERR(srv->cls);
 
-- 
2.46.0

