From f621cf43274a3ba9a644ed02cd38f09919244e41 Mon Sep 17 00:00:00 2001
From: jjm2473 <jjm2473@gmail.com>
Date: Wed, 3 Sep 2025 15:05:32 +0800
Subject: [PATCH] fix rockchip_opp_set_low_length undefined

---
 include/soc/rockchip/rockchip_opp_select.h | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/include/soc/rockchip/rockchip_opp_select.h
+++ b/include/soc/rockchip/rockchip_opp_select.h
@@ -166,6 +166,8 @@ int rockchip_set_intermediate_rate(struc
 				   struct clk *clk, unsigned long old_freq,
 				   unsigned long new_freq, bool is_scaling_up,
 				   bool is_set_clk);
+int rockchip_opp_set_low_length(struct device *dev, struct device_node *np,
+				struct rockchip_opp_info *opp_info);
 int rockchip_opp_config_regulators(struct device *dev,
 				     struct dev_pm_opp *old_opp,
 				     struct dev_pm_opp *new_opp,
@@ -252,6 +254,13 @@ rockchip_set_intermediate_rate(struct de
 {
 	return -EOPNOTSUPP;
 }
+
+static inline int
+rockchip_opp_set_low_length(struct device *dev, struct device_node *np,
+			    struct rockchip_opp_info *opp_info)
+{
+	return -EOPNOTSUPP;
+}
 
 static inline int
 rockchip_opp_config_regulators(struct device *dev,
