From afa7e4065ba4f577a516feeca495e4ecc527524e Mon Sep 17 00:00:00 2001
From: jjm2473 <jjm2473@gmail.com>
Date: Wed, 10 Sep 2025 13:31:37 +0800
Subject: [PATCH] include/rockchip_sip.h: sync from develop-6.6

---
 include/linux/rockchip/rockchip_sip.h | 56 +++++++++++++++++++++++++++
 1 file changed, 56 insertions(+)

--- a/include/linux/rockchip/rockchip_sip.h
+++ b/include/linux/rockchip/rockchip_sip.h
@@ -33,6 +33,7 @@
 #define PSCI_SIP_VPU_RESET		0x8200000c
 #define SIP_BUS_CFG			0x8200000d
 #define SIP_LAST_LOG			0x8200000e
+#define SIP_ACCESS_MEM_OS_REG		0x8200000f
 #define SIP_SCMI_AGENT0			0x82000010
 #define SIP_SCMI_AGENT1			0x82000011
 #define SIP_SCMI_AGENT2			0x82000012
@@ -58,6 +59,8 @@
 #define SIP_HDMIRX_CFG			0x82000027
 #define SIP_MCU_CFG			0x82000028
 #define SIP_PVTPLL_CFG			0x82000029
+#define SIP_GPIO_CFG			0x8200002c
+#define SIP_CPU_PM_CFG			0x8200002d
 
 #define TRUSTED_OS_HDCPKEY_INIT		0xB7000003
 
@@ -110,6 +113,13 @@
 #define LINUX_PM_STATE			0x09
 #define SUSPEND_IO_RET_CONFIG		0x0a
 #define SLEEP_PIN_CONFIG		0x0b
+#define SLEEP_IO_CONFIG			0x0c
+
+enum {
+	RK_PM_SLEEP_IO_CFG_CNT = 0,
+	RK_PM_SLEEP_IO_CFG_VAL = 1,
+	RK_PM_SLEEP_IO_CFG_MAX,
+};
 
 /* SIP_REMOTECTL_CFG call types */
 #define	REMOTECTL_SET_IRQ		0xf0
@@ -120,6 +130,12 @@
 /* wakeup state */
 #define REMOTECTL_PWRKEY_WAKEUP		0xdeadbeaf
 
+/* SIP_ACCESS_MEM_OS_REG child configs */
+enum {
+	RK_MEM_OS_REG_READ = 0,
+	RK_MEM_OS_REG_WRITE,
+};
+
 /* SIP_MCU_CFG child configs, MCU ID */
 enum {
 	RK_BUS_MCU,
@@ -232,6 +248,22 @@ enum {
 	PVTPLL_GET_INFO = 0,
 	PVTPLL_ADJUST_TABLE = 1,
 	PVTPLL_LOW_TEMP = 2,
+	PVTPLL_VOLT_SEL = 3,
+};
+
+/* SIP_GPIO_CFG child configs */
+enum {
+	GPIO_GET_GROUP_INFO = 0,
+	GPIO_SET_GROUP_INFO = 1,
+	GPIO_GET_VIRT_EN = 2,
+	GPIO_SET_VIRT_EN = 3,
+	GPIO_SET_STORE_ST = 4,
+	GPIO_CLEAR_STORE_ST = 5,
+};
+
+/* SIP_CPU_PM_CFG child configs */
+enum {
+	CPU_PM_CLUST_AUTO_PD_EN = 0,
 };
 
 struct pt_regs;
@@ -262,6 +294,7 @@ int sip_smc_secure_reg_write(u32 addr_ph
 u32 sip_smc_secure_reg_read(u32 addr_phy);
 struct arm_smccc_res sip_smc_bus_config(u32 arg0, u32 arg1, u32 arg2);
 struct dram_addrmap_info *sip_smc_get_dram_map(void);
+int sip_smc_access_mem_os_reg(u32 func, u32 id, u32 *val);
 int sip_smc_amp_config(u32 sub_func_id, u32 arg1, u32 arg2, u32 arg3);
 struct arm_smccc_res sip_smc_get_amp_info(u32 sub_func_id, u32 arg1);
 struct arm_smccc_res sip_smc_get_pvtpll_info(u32 sub_func_id, u32 arg1);
@@ -270,6 +303,9 @@ struct arm_smccc_res sip_smc_pvtpll_conf
 
 void __iomem *sip_hdcp_request_share_memory(int id);
 struct arm_smccc_res sip_hdcp_config(u32 arg0, u32 arg1, u32 arg2);
+struct arm_smccc_res sip_smc_gpio_config(u32 sub_func_id, u32 arg1, u32 arg2,
+					 u32 arg3);
+int sip_smc_cpu_pm_config(u32 func, u32 id, u32 cfg);
 ulong sip_cpu_logical_map_mpidr(u32 cpu);
 /***************************fiq debugger **************************************/
 void sip_fiq_debugger_enable_fiq(bool enable, uint32_t tgt_cpu);
@@ -360,6 +396,13 @@ static inline struct dram_addrmap_info *
 	return NULL;
 }
 
+static inline int sip_smc_access_mem_os_reg(u32 func,
+					    u32 id,
+					    u32 *val)
+{
+	return 0;
+}
+
 static inline int sip_smc_amp_config(u32 sub_func_id,
 				     u32 arg1,
 				     u32 arg2,
@@ -406,6 +449,19 @@ static inline struct arm_smccc_res sip_h
 	return tmp;
 }
 
+static inline struct arm_smccc_res sip_smc_gpio_config(u32 sub_func_id, u32 arg1,
+						       u32 arg2, u32 arg3)
+{
+	struct arm_smccc_res tmp = { .a0 = SIP_RET_NOT_SUPPORTED, };
+
+	return tmp;
+}
+
+static inline int sip_smc_cpu_pm_config(u32 func, u32 id, u32 cfg)
+{
+	return SIP_RET_NOT_SUPPORTED;
+}
+
 static inline ulong sip_cpu_logical_map_mpidr(u32 cpu) { return 0; }
 
 /***************************fiq debugger **************************************/
