From 98feb9ac47f8aeae0e7d8cbe911ff56c27fa4f1e Mon Sep 17 00:00:00 2001
From: jjm2473 <jjm2473@gmail.com>
Date: Wed, 10 Sep 2025 13:32:17 +0800
Subject: [PATCH] rknpu_devfreq: sync from develop-6.6

---
 drivers/rknpu/rknpu_devfreq.c | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

--- a/drivers/rknpu/rknpu_devfreq.c
+++ b/drivers/rknpu/rknpu_devfreq.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright (C) Rockchip Electronics Co.Ltd
+ * Copyright (C) Rockchip Electronics Co., Ltd.
  * Author: Finley Xiao <finley.xiao@rock-chips.com>
  */
 
@@ -120,6 +120,15 @@ static int rk3588_npu_get_soc_info(struc
 		else if (value == 0xa)
 			*bin = 2;
 	}
+	if (of_property_match_string(np, "nvmem-cell-names", "customer_demand") >= 0) {
+		ret = rockchip_nvmem_cell_read_u8(np, "customer_demand", &value);
+		if (ret) {
+			dev_err(dev, "Failed to get customer_demand\n");
+			return ret;
+		}
+		if (value == 0x3)
+			*bin = 4;
+	}
 	if (*bin < 0)
 		*bin = 0;
 	LOG_DEV_INFO(dev, "bin=%d\n", *bin);
@@ -264,6 +273,10 @@ static const struct of_device_id rockchi
 		.data = (void *)&rk3576_npu_opp_data,
 	},
 	{
+		.compatible = "rockchip,rk3576s",
+		.data = (void *)&rk3576_npu_opp_data,
+	},
+	{
 		.compatible = "rockchip,rk3588",
 		.data = (void *)&rk3588_npu_opp_data,
 	},
@@ -422,7 +435,7 @@ int rknpu_devfreq_runtime_suspend(struct
 	struct rknpu_device *rknpu_dev = dev_get_drvdata(dev);
 	struct rockchip_opp_info *opp_info = &rknpu_dev->opp_info;
 
-	if (opp_info->is_scmi_clk) {
+	if (rockchip_opp_is_use_pvtpll(opp_info)) {
 		if (clk_set_rate(opp_info->clk, POWER_DOWN_FREQ))
 			LOG_DEV_ERROR(dev, "failed to restore clk rate\n");
 	}
@@ -450,7 +463,7 @@ int rknpu_devfreq_runtime_resume(struct
 	if (opp_info->data && opp_info->data->set_read_margin)
 		opp_info->data->set_read_margin(dev, opp_info,
 						opp_info->target_rm);
-	if (opp_info->is_scmi_clk) {
+	if (rockchip_opp_is_use_pvtpll(opp_info)) {
 		if (clk_set_rate(opp_info->clk, rknpu_dev->current_freq))
 			LOG_DEV_ERROR(dev, "failed to set power down rate\n");
 	}
